File: leetcode-python-go/1658. 将 x 减到 0 的最小操作数/main.go
package main

import (
	"fmt"
)

func minOperations(nums []int, x int) int {
	n := len(nums)
	sum := 0
	for _, num := range nums {
		sum += num
	}
	if sum < x {
		return -1
	}

	right := 0
	lsum := 0
	rsum := sum
	ans := n + 1

	for left := -1; left < n; left++ {
		if left != -1 {
			lsum += nums[left]
		}
		for right < n && lsum+rsum > x {
			rsum -= nums[right]
			right++
		}
		if lsum+rsum == x {
			ans = min(ans, (left+1)+(n-right))
		}
	}
	if ans > n {
		return -1
	}
	return ans
}

func min(a, b int) int {
	if b < a {
		return b
	}
	return a
}

func main() {
	nums := []int{5, 6, 7, 8, 9}
	x := 4
	results := minOperations(nums, x)
	fmt.Println(results)
}

File: leetcode-python-go/1658. 将 x 减到 0 的最小操作数/main.py
from typing import List

nums = [1,1,4,2,3]
x = 5

def minOperations(nums: List[int], x: int) -> int:
    n = len(nums)
    total = sum(nums)

    if total < x:
        return -1
    
    right = 0
    lsum, rsum = 0, total
    ans = n + 1
    for left in range(-1, n - 1):
        if left != -1:
            lsum += nums[left]
        while right < n and lsum + rsum > x:
            rsum -= nums[right]
            right += 1
        if lsum + rsum == x:
            ans = min(ans, (left + 1) + (n - right))
    
    return -1 if ans > n else ans

print(minOperations(nums, x))
